compiler/src/dotty/tools/dotc/reporting/WConf.scala

```diff
@@ -21,11 +20,15 @@ enum MessageFilter:
       val noHighlight = message.msg.message.replaceAll("\\e\\[[\\d;]*[^\\d;]","")
       pattern.findFirstIn(noHighlight).nonEmpty
     case MessageID(errorId) => message.msg.errorId == errorId
+    case SourcePattern(pattern) =>
+      val path = message.position.orElse(NoSourcePosition).source().path()
+      pattern.findFirstIn(path).nonEmpty
     case None => false
 
   case Any, Deprecated, Feature, Unchecked, None
   case MessagePattern(pattern: Regex)
   case MessageID(errorId: ErrorMessageID)
+  case SourcePattern(pattern: Regex)
```

Comment:
Old code had a cache, shouldn't that be provided too?

Reply:
cache added